### DISCUSSION TITLE
Stock Revision Control

### DISCUSSION THREAD

**Discussion by bberg115** [inventree/InvenTree#5930] [Q&A] [answered]
I am not sure if we're using it incorrectly or if the feature doesn't exist yet, but we're running into an issue with revisions of stock. How does Inventree distinguish revisions of a part when it comes to stock?  If this isn't implemented, is there a work around or plan to address it?

It appears that changing the revision of the base part does not effect existing or future stock.

Background: We revised a part from Rev A to Rev B and noticed that existing stock now states the base part as Rev B.

1. API Info does not state anything with respect to revision
{"batch":"","belongs_to":null,"build":null,"consumed_by":null,"customer":null,"delete_on_deplete":false,"expiry_date":null,"is_building":false,"link":"","location":15,"notes":null,"owner":null,"packaging":null,"part":743,"purchase_order":131,"purchase_order_reference":"PO-00131","pk":1663,"quantity":25.0,"sales_order":null,"serial":null,"status":10,"status_text":"OK","stocktake_date":null,"supplier_part":512,"barcode_hash":"","updated":"2023-10-10 16:16","purchase_price":"18.050000","purchase_price_currency":"USD","allocated":0.0,"expired":false,"installed_items":0,"stale":false,"tracking_items":1,"tags":[]},

**Comment by matmair** [comment, marked as answer]
Hi there @bberg115! The revision field is just a data value and has no impact on business logic - it is basically like a more specific description.
If you change it on a part, the existing (and future) stock will reflect that change.

**Comment by bberg115** [comment]
Thanks for the response. Albeit a low priority, it would be nice to maintain revision of stock to help with isolate revisions as changes are made. Is that something that could be "easily" implemented?

We'll work around this by manually adjusting stock as revisions occur. Cheers!

**Reply by matmair** [reply]
Maybe the communication around the field could be improved with an info text. Stock is always bound to a specific part and new revisions should always be a new part so that info should already be tracked - or am I misunderstanding your request?

**Reply by jesseyu1984918** [reply]
Sometimes we'd wish to avoid duplicate part only because of reivsion, ideally revision can be a pull down control and all the parameters would be pulled while the part itself is not changed, this would probably would be more of a UI thing and not related to database because by right the new revision should be a new entry anyway.

**Reply by bberg115** [reply]
@matmair sorry, I was offline last week. You understand my issue correctly, I think that is a reasonable solution and agree with Jessey on the GUI updates in the future.
